views/tx.tmpl

```diff
@@ -279,6 +282,59 @@
                   </td>
                 </tr>
               {{end}}
+              {{range $.Swaps}}
+                {{if eq .ContractTxID $.Data.TxID}}
+                <!-- display redemption tx id if this tx is the contract tx -->
+                <tr>
+                  <td class="text-right medium-sans text-nowrap pr-2 py-2">
+                    Redemption Tx:
+                  </td>
+                  <td class="text-left py-1 text-secondary clipboard" colspan="3">
+                    {{template "hashElide" (hashlink .RedemptionTxID (print "/tx/" .RedemptionTxID))}}
+                  </td>
+                </tr>
+                {{else}}
+                <!-- display contract tx id if this tx is the redemption/refund tx -->
+                <tr>
+                  <td class="text-right medium-sans text-nowrap pr-2 py-2">
+                    Contract Tx:
+                  </td>
+                  <td class="text-left py-1 text-secondary clipboard" colspan="3">
+                    {{template "hashElide" (hashlink .ContractTxID (print "/tx/" .ContractTxID))}}
+                  </td>
+                </tr>
+                {{end}}
+                <tr>
+                  <td class="text-right medium-sans text-nowrap pr-2 py-2">
+                    Redeemed By:
+                  </td>
+                  <td class="text-left py-1 text-secondary">
+                    {{.RedeemedBy}}
+                  </td>
+                  <td class="text-right medium-sans text-nowrap pr-2 py-2">
+                    Redeemed To:
+                  </td>
+                  <td class="text-left py-1 text-secondary clipboard">
+                    {{range .RedeemedToAddresses}}
+                      {{template "hashElide" (hashlink . (print "/address/" .))}}
+                    {{end}}
+                  </td>
+                </tr>
+                <tr>
+                  <td class="text-right medium-sans text-nowrap pr-2 py-2">
+                    Contract Value:
+                  </td>
+                  <td class="text-left py-1 text-secondary">
+                    {{.FormattedAmount}}
+                  </td>
+                  <td class="text-right medium-sans text-nowrap pr-2 py-2">
+                    Lock Time:
+                  </td>
+                  <td class="text-left py-1 text-secondary clipboard">
+                    {{.Locktime}}
+                  </td>
+                </tr>
+              {{end}}
             </tbody>
```

Comment:
Since there can and will be multiple redemptions or swaps in a single transaction, the best place to display this info is in the input and output tables. 

In the output table, maybe use the `Type` column to say **swap** when the output is a swap output.

For the inputs, it's a little harder, because we are already tight on space in that table. I would be ok with not displaying the `txid:vout` directly for redemptions and refunds, and instead just displaying a link with the text **redemption** or **refund**. The txid and vout detail would still be visible on hover, and could also be put in a tooltip. Thoughts @chappjc?

Reply:
Tables something like

![image](https://user-images.githubusercontent.com/6109680/80833766-e163a700-8bb4-11ea-93f4-ba261ebf6524.png)
 
Though now that I'm looking at it, "Swap Refund" doesn't really work there. Just **Swap** would be better.

Reply:
Re the inputs table:
Since the **Previous Outpoint** column links to the output that was spent, shouldn't this be `Contract` or `Swap Contract`? The spent output for a redeem/refund tx input would be a contract output, not a refund or redemption...